### PR TITLE
webpack

### PR DESCRIPTION
**Pull request by StephenWeatherford**
NOTES:
1) Might be more speed-up to be obtained here by reducing contents of externalNodeModules (update: cold start: 3.5s, warm start: 300ms, installation time somewhat longer than Docker, but all good enough)
2) Might change the way we're dealing with resources in vscode-azureextensionui, but this does work and I thought it might be similar to what we'd do for third party code.  That remains to be seen.
3) Intend to move some of the webpack.config.js code to our shared dev package

### CODE REVIEW

File: entrypoint.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+Object.defineProperty(exports, "__esModule", { value: true });
+
+const extension = require("./dist/extension");
+
+async function activate(ctx) {
+    return await extension.activateInternal(ctx, perfStats);
+}
+
+exports.activate = activate;
```
Comment (PrashanthCorp) [inline entrypoint.js]: I don't see this being imported anywhere. Are these used by AzureTools?

Comment (StephenWeatherford) [reply entrypoint.js]: Yes.  Added comment.

Comment (PrashanthCorp) [reply entrypoint.js]: Ah, sounds good.

File: extension.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export { parseDocDBConnectionString } from './src/docdb/docDBConnectionStrings';
+export { emulatorPassword } from './src/constants';
+export { getDocumentTreeItemLabel } from './src/utils/vscodeUtils';
+export { validOnTimeoutOrException } from './src/utils/inputValidation';
+export { addDatabaseToAccountConnectionString, getDatabaseNameFromConnectionString } from './src/mongo/mongoConnectionStrings';
+export { MongoCommand } from './src/mongo/MongoCommand';
+export { getAllCommandsFromText, getCommandFromTextAtLocation } from './src/mongo/MongoScrapbook';
+export { rejectOnTimeout, valueOnTimeout } from './src/utils/timeout';
+
+// These use instanceof and therefore we need to make sure we're using the same version of the bson module in the tests as in the bundle,
```
Comment (EricJizbaMSFT) [inline extension.ts]: Does this only apply to test code? If not, it makes me kinda nervous. What exactly is the problem with instanceof? I guess I'd rather find a fix for instanceof than single out every single use of it we run in to.

Is this at all related to the thing you did here? https://github.com/Microsoft/vscode-docker/pull/745 Either way don't forget to do that here :)

File: package-lock.json [hunk 3]
```diff
@@ -126,9 +126,9 @@
 			}
```
Comment (EricJizbaMSFT) [inline package-lock.json]: Can you run another update on this file? I still see some changes when I run `npm install` on my machine. (typescript changed from 2.9.2 to 3.2.2)

File: src/extensionVariables.ts [hunk 4]
```diff
@@ -17,6 +18,7 @@ export namespace ext {
     export let context: ExtensionContext;
     export let outputChannel: OutputChannel;
     export let reporter: ITelemetryReporter;
+    export let resourcesPath = path.join(__dirname, 'resources');
```
Comment (EricJizbaMSFT) [inline src/extensionVariables.ts]: You have `resourcesPath` in both constants and extensionVariables. Do we need both? Seems kinda confusing.

File: extension.ts (hunk 2 above)
Comment (StephenWeatherford) [reply extension.ts]: Yes, tests only (updated comment).  The issue is that the test code is completely separate from the extension bundle, so if we do a blind import from the tests, you get two different instances of the module and hence two difference classes with different identity.  That's also why we're importing everything from extension.ts instead of importing individual files directly from the tests like we used to, because you'd end up with two different instances of the code (and hence extensionVariables isn't a singleton, etc.).  

extension.js is separate from the test code and uses an internal webpack version of import, so it doesn't know about anything imported by code outside of it (something I wish we could change - I tried).

Comment (StephenWeatherford) [reply extension.ts]: > Is this at all related to the thing you did here? Microsoft/vscode-docker#745 Either way don't forget to do that here :)

No.  The other fix will be in the shared version.

File: src/extensionVariables.ts (hunk 4 above)
Comment (StephenWeatherford) [reply src/extensionVariables.ts]: Removed from extvars
